compiler/src/main/resources/Dubbo3InterfaceStub.mustache

```diff
@@ -18,11 +18,37 @@ static final String SERVICE_NAME = "{{commonPackageName}}.{{serviceName}}";
     // FIXME, initialize Dubbo3 stub when interface loaded, thinking of new ways doing this.
     static final boolean inited = {{className}}.init();
 
-{{#methods}}
+{{#unaryMethods}}
+    {{#javaDoc}}
+        {{{javaDoc}}}
+    {{/javaDoc}}
+    {{#deprecated}}
+        @java.lang.Deprecated
+    {{/deprecated}}
     {{outputType}} {{methodName}}({{inputType}} request);
 
     CompletableFuture<{{outputType}}> {{methodName}}Async({{inputType}} request);
 
-{{/methods}}
+{{/unaryMethods}}
+
+{{#serverStreamingMethods}}
+    {{#javaDoc}}
+        {{{javaDoc}}}
+    {{/javaDoc}}
+    {{#deprecated}}
+        @java.lang.Deprecated
+    {{/deprecated}}
+    void {{methodName}}({{inputType}} request, org.apache.dubbo.common.stream.StreamObserver<{{outputType}}> responseObserver);
+{{/serverStreamingMethods}}
+
+{{#biStreamingMethods}}
+    {{#javaDoc}}
+        {{{javaDoc}}}
+    {{/javaDoc}}
+    {{#deprecated}}
+        @java.lang.Deprecated
+    {{/deprecated}}
+    org.apache.dubbo.common.stream.StreamObserver<{{inputType}}> {{methodName}}(org.apache.dubbo.common.stream.StreamObserver<{{outputType}}> responseObserver);
+{{/biStreamingMethods}}
```

Comment:
I think we should consider how the developers will use this before determining support of streaming interface:
* For the consumer side, developers can call any of the `sync`, `async`, `streaming` methods from the proxy they get.
* For the provider side, developers don't have to provider implementation for all methods, one bi-directional streaming implementation would be enough.

Reply:
i think we should support like grpc. That's enough.This is also how tri Stream is implemented

Reply:
Yes, but simply add those streaming method declarations will be a burden for the Provider, we should think of letting the Provider only care about the bi-directional streaming implementation.

Reply:
not have to give the implementation of all methods.

Reply:
server stream and bi stream is diff, Developers should decide which to use

Reply:
They are different and that's the key point we need to determine. I think for the most generic case, Provider can provider one streaming implementation that can serve all different kinds of calls from the Consumer side.

Reply:
Sure we can let the Provider decide to only support unary or streaming or bi streaming, but is that necessary, or at least whey they want to choose they don't need to give the implementation of all kinds.

Reply:
You mean to provide a default implementation that wraps the service stream with a bi-directional stream